Allows to pass ajax options to nmModal() using option ajax

arunpjohny opened the pull request:
Issues faced:
    Needed an option to pass the `dataType` option to the ajax request
    Needed an option to pass some custom `data` to the ajax request
    Needed an option to pass the request `type` to the ajax requst

If the ajax `dataType` option is overriden to `json` using http://api.jquery.com/jQuery.ajaxSetup/ then all attempts to load nyroModal with ajax contents fails.

Changes in both link and form filters to allow the additional option `ajax`.

Usage:

```
$('#manual1').click(function(e) {
    e.preventDefault();
    $.nmManual('demoSent.php', {
        ajax: {
            data: {
                myparam: "one"
            },
            type: "POST",
            dataType: "text"
        }
    });
});
```

arunpjohny committed:
Allow to set additional ajax options while using nmModal().

If we override the default ajax dataType as `json` or anything else then the ajax request using nmModal() fails
Also helps to change the request type or any other properties other than url, success and failure

arunpjohny commented on js/jquery.nyroModal.filters.link.js:
```diff
@@ -24,16 +24,18 @@ jQuery(function($, undefined) {
+						}
+					});
+
+				$.ajax(ajax);
```
When the nmModal() call is made an option `ajax` can be passed to it. It can override all `jQuery.ajax()` options other than url, success and failure.

The default behavior of the data parameter is retained while appending the values passed to it via the `ajax` option.

Usage:

```
$('#manual1').click(function(e) {
    e.preventDefault();
    $.nmManual('demoSent.php', {
        ajax: {
            data: {
                myparam: "one"
            },
            type: "POST",
            dataType: "text"
        }
    });
});
```

nyroDev commented:
Nice pull request. Could you also add the same update on the form filter? Then I will accept your pull request on the main repistory.
Thanks.

arunpjohny committed:
Allow to set additional ajax options while using nmModal() with form filter.

    If we override the default ajax dataType as `json` or anything else then the ajax request using nmModal() fails
    Also helps to change the request type or any other properties other than url, success and failure
